boards/sim/sim/sim/scripts/Make.defs

```diff
@@ -259,6 +259,9 @@ LDELFFLAGS += -T $(call CONVERT_PATH,$(TOPDIR)/binfmt/libelf/gnu-elf.ld)
 ifeq ($(CONFIG_HOST_MACOS),y)
   LDFLAGS += -Wl,-dead_strip
 else
+  # Let the symbol table link from 0x400000
+  # which is more convenient for debugging.
+  LDFLAGS += -Wl,-Ttext-segment=0x400000
   LDFLAGS += -Wl,--gc-sections
```

Comment:
```suggestion
  LDFLAGS += -Wl,--gc-sections
  
  # Let the symbol table link from 0x400000
  # which is more convenient for debugging.
  LDFLAGS += -Wl,-Ttext-segment=0x400000
```

Just to have better visibility that `LDFLAGS += -Wl,-dead_strip` is same as `LDFLAGS += -Wl,--gc-sections` and comment section is not related to `LDFLAGS += -Wl,--gc-sections`